euphonic/force_constants.py

```diff
@@ -938,18 +933,20 @@ def _calculate_dyn_mat(
         full_dyn_mat = fc_img_weighted*(
             ij_phases.repeat(3, axis=2).repeat(3, axis=1))
         dyn_mat = np.sum(full_dyn_mat, axis=0)
-        if len(all_origins_cart) > 0:
-            all_phases = np.einsum('ijkl,i->ijkl',
-                                   sc_phases[sc_image_i], cell_phases)
-            r_vec_sum = 1j*np.einsum('ijkl,ijklm->ijkm',
-                                     all_phases, all_origins_cart)
-            dmat_gradient = fc_img_weighted[..., ax]*(r_vec_sum.repeat(
-                3, axis=2).repeat(3, axis=1))
-            dmat_gradient = np.sum(dmat_gradient, axis=0)
-            return dyn_mat, dmat_gradient
-        else:
+
+        if len(all_origins_cart) == 0:

```

Comment:
```suggestion
        if not all_origins_cart:
```

Reply:
Will need to revert this one, it's raising errors:

```
2025-03-20T17:02:54.2280270Z >       if not all_origins_cart:
2025-03-20T17:02:54.2281007Z E       ValueError: The truth value of an empty array is ambiguous. Use `array.size > 0` to check that an array is not empty.
```